modules/dispatcher/dispatch.c

```diff
@@ -1030,7 +1027,7 @@ static ds_data_t* ds_load_data(ds_partition_t *partition, int use_state_col)
 		weight = 1;
 
 		/* weight */
-		if (values[3].type == DB_INT) {
+		if (VAL_TYPE(values+3) == DB_INT || VAL_TYPE(values+3) == DB_BIGINT) {
```

Comment:
Why not using `get_int_from_dbval` here also?

Reply:
Hello Bogdan,

I think, the weight column should have integer or char (**freeswitch esl string for dynamic weight calculation**) datatype. so I just provide the flexibility there. But as per our mysql table version the weight column has char datatype. If we want to use only char datatype there, then we can even remove even integer support for the weight column. what you thing about that?